### PR TITLE
Turns bat knockback into an element and adds it to cleric maces

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Turns bat knockback into an element and adds it to cleric maces.

To farm GBP, I also turned the knockback into a standard element.

## Why It's Good For The Game

Cleric maces are pretty poop. Ruin-only, bulky, and with only slightly above average damage, there's little point to hauling one around most of the time as opposed to a welding tool or something. I saw the description referenced the baseball bat, but it doesn't actually the bat's knockback gimmick, which I think would be pretty damn useful for it to be 1. less useless and 2. more useful for the space explorers that will find it.

> To farm GBP, I also turned the knockback into a standard element.

Now it can be added to other things yayyyyyyyyy

## Changelog

:cl:
refactor: Turns baseball bat knockback into an element
add: and adds it to cleric maces
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
People stealing items to be able to see in the dark is preferable to them being born that way

**Comment by carlarctg** [comment]
> well, there's also the disk that can print cleric maces via autolathe, and using titanium as the material gives it good AP and 7.8 hits to down iirc

5.6 and the disk is found where the mace is

**Comment by PKPenguin321** [comment]
Make the distance/speed it throws you admin editable please :)

File: code/datums/elements/attack_knockback.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Used in bats and maces
+ */
+/datum/element/attack_knockback
+
+/datum/element/attack_knockback/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_ITEM_ATTACK, PROC_REF(on_item_attack))
```
Comment (MrMelbert) [inline code/datums/elements/attack_knockback.dm:12]: Should really be using afterattack signal (just make sure to check for proximity)

`COMSIG_ITEM_ATTACK` is sent before pacifism and nobludgeon checks which means this signal may be caught even if an "attack" didn't "happen'

**Comment by Potato-Masher** [comment]
Does this touch the knockback component at all

**Comment by carlarctg** [comment]
> Does this do anything with the knockback component at all or is this separate

there is no such thnig

**Comment by ZephyrTFA** [comment]
don't forget about mjollnir

**Comment by carlarctg** [comment]
> don't forget about mjollnir

angry facef

**Comment by carlarctg** [comment]
> don't forget about mjollnir

Mjolnir uses different, more elaborate code. I am declaring it out of scope

**Comment by ZephyrTFA** [comment]
it's not out of scope.
itr is a knock back effect with damage

**Comment by carlarctg** [comment]
>:(

File: code/datums/elements/attack_knockback.dm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	/// Picks a number from this list, throws this far away
+	var/list/throw_range = list(1)
+	/// Picks a number from this list, throws at this speed
+	var/list/throw_speed = list(1, 4)
+	/// If the throw is violent or not
+	var/gentle = TRUE
+	/// Callback for a proc right before confirming the attack. If it returns FALSE, cancel
+	var/datum/callback/pre_hit_callback
```
Comment (Jacquerel) [inline code/datums/elements/attack_knockback.dm:16]: Are you going to fix this or not, we already told you how on Discord

**Comment by Jacquerel** [comment]
Reopen this when you want to work on it again

**Pull request closed without merging**